routes/users.js

```diff
@@ -211,20 +216,27 @@ router.post('/signup', redirectProfile, async (req, res) => {
 	} else {
 		const salt = await bcrypt.genSalt(10);
 		const hashedPassword = await bcrypt.hash(password, salt);
+		const verificationHash = createEmailVerificationHash();
+
 		const newUser = new User({
 			fName,
 			lName,
 			email,
+			verificationHash,
 			password: hashedPassword,
 			userRole
 		});
-		var userId = mongoose.Types.ObjectId(newUser._id);
-		req.session.userId = userId;
+		req.session.userId = mongoose.Types.ObjectId(newUser._id);
 		try {
 			await newUser.save();
 			//trying to add a second step here
 			//if the userRole is partner then redirect to agency.ejs then profile.ejs
-			if (newUser.userRole == 'partner') {
+
+			const emailResponse = await sendMail('gnorbsl@gmail.com', email, 'Email verification', 'Please verify your email by clicking on this link: http://localhost:8081/users/verify/' + verificationHash)
```

Comment:
Don't forget to swap your email for user's email :)

Reply:
@gnorbsl @Jacobjeevan ^ what Jacob said